src/Microsoft.OData.Core/UriParser/SemanticAst/SelectExpandClauseExtensions.cs

```diff
@@ -157,28 +172,48 @@ private static string BuildTopLevelSelect(SelectExpandClause selectExpandClause)
         /// </summary>
         /// <param name="selectedItem">the select item to translate</param>
         /// <returns>the string representation of this select item, or null if the select item is an expandedNavPropSelectItem</returns>
-        private static string GetSelectString(SelectItem selectedItem)
+        private static IList<string> GetSelectString(SelectItem selectedItem)
         {
             WildcardSelectItem wildcardSelect = selectedItem as WildcardSelectItem;
             NamespaceQualifiedWildcardSelectItem namespaceQualifiedWildcard = selectedItem as NamespaceQualifiedWildcardSelectItem;
             PathSelectItem pathSelectItem = selectedItem as PathSelectItem;
 
-            if (wildcardSelect != null)
+            if (wildcardSelect!= null)
             {
-                return "*";
+                return new List<string>() { "*" };
             }
             else if (namespaceQualifiedWildcard != null)
             {
-                return namespaceQualifiedWildcard.Namespace + ".*";
+                return new List<string>() { namespaceQualifiedWildcard.Namespace + ".*" };
             }
             else if (pathSelectItem != null)
             {
-                return String.Join("/", pathSelectItem.SelectedPath.WalkWith(PathSegmentToStringTranslator.Instance).ToArray());
-            }
-            else
-            {
-                return null;
+                IList<string> nextLevelSelectList = null;
+                if (pathSelectItem.SelectAndExpand != null)
+                {
+                    nextLevelSelectList = GetCurrentLevelSelectList(pathSelectItem.SelectAndExpand);
+                }
+
+                var selectListItem = String.Join("/", pathSelectItem.SelectedPath.WalkWith(PathSegmentToStringTranslator.Instance).ToArray());

```

Comment:
>ToArray [](start = 131, length = 7)

why call "ToArray()"

Reply:
Makes sense string.Join takes Ienumerable<string> as well.  There was ToArray in the existing code, so kept it. Now realized walkwith returns Ienumerable<string>

Reply:
REsolved

Reply:
Build failed since it accepts only string array for older frameworks , so putting .ToArray back